### PR TITLE
[#21] Enhance README

### LINKED ISSUE #21: Write good readme and maybe even tutorial
Current README doesn't explains how to use `log-warper` which is bad. This README should be written. Too much effort was put into `log-warper`. So we must try to make this library more production-ready.

### PR DESCRIPTION
**Pull request by chshersh**
Resolves #21

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,21 +7,98 @@ log-warper
...
-2. Supports logging initialization from `.yaml` configuration file
-3. Has wrapper for pure logging via `StateT`
-4. Supports log rotation
-5. Flexibles and easy setting up of loggers (using monoidal builders and lenses)
-6. Ability to acquire last `N` megabytes of logs from in-memory cache
+## Features
 
-Contributing
-------------
+Here is the list of feature `log-warper` provides.
```
Comment (vrom911) [inline README.md]: Small typo, "list of feature**s**"

File: README.md [hunk 2]
```diff
@@ -7,21 +7,98 @@ log-warper
...
+
+1. Hierarchical logger names.
+
+   Logger names (tags for loggers) forms hierarchy. It means, that `""`
+   (also known as `mempty` or `rootLoggerName`) is a parent of logger with name `"node"` which isDir
```
Comment (vrom911) [inline README.md]: `isDir` — auto completion mistake? :)

File: README.md [hunk 3]
```diff
@@ -7,21 +7,98 @@ log-warper
...
+   a parent of logger with name `"node.communication"`. So, logger name comprises dot-separated components.
+   This means that if some logger name doesn't have some settings (like severity or output file) it takes
+   its settings from its closest parent, containing this settings.
```
Comment (vrom911) [inline README.md]: twice "its" doesn't sound cool, probably "takes its settings form *the* closest parent" or vice versa

File: README.md [hunk 4]
```diff
@@ -7,21 +7,98 @@ log-warper
...
+   ```
+
+5. How can I enable messages with severity `Info` and higher?
+
+   * Write `severity: Info+` inside tree node of your logger settings.
+
+6. How can I log inside functions like `forkIO`?
+
+   * Use [`liftLogIO`](https://hackage.haskell.org/package/log-warper-1.8.5/docs/System-Wlog-CanLog.html#v:liftLogIO) function.
+   It's Haddock contains nice usage example.
```
Comment (vrom911) [inline README.md]: Small typo, "Its"

**Review by vrom911 — COMMENTED** [review]
Great! It looks better now, and all topics seem to be covered! :1st_place_medal:

File: README.md [hunk 5]
```diff
@@ -7,21 +7,98 @@ log-warper
...
-6. Ability to acquire last `N` megabytes of logs from in-memory cache
+## Features
 
-Contributing
-------------
+Here is the list of feature `log-warper` provides.
+
+1. Hierarchical logger names.
+
+   Logger names (tags for loggers) forms hierarchy. It means, that `""`
```
Comment (gromakovsky) [inline README.md]: s/forms/form

**Review by vrom911 — APPROVED** [review]

**Pull request merged**